scripts/identify-web-features.js

```diff
@@ -0,0 +1,303 @@
+import { Octokit } from "octokit";
+import { features } from "web-features";
+import yargs from "yargs";
+
+// This is used as a hidden HTML comment when posting comments to GitHub issues.
+// This way, we can retrieve the comment later, and update it if needed.
+const HIDDEN_COMMENT_IN_ISSUE = "<!-- interop-proposals-bot web-features update -->";
+const GITHUB_API_VERSION = "2022-11-28";
+
+const argv = yargs(process.argv)
+  .option("number", {
+    alias: "n",
+    type: "number",
+    default: false,
+    describe: "The issue number to process",
+  })
+  .option("repo", {
+    alias: "r",
+    type: "string",
+    describe: "The owner and repository name. For example: web-platform-tests/interop",
+  }).argv;
+
+const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+
+function escapeFeatureName(feature) {
+  // Escape the feature name for use in HTML.
+  return feature.name.replace(/</g, "&lt;").replace(/>/g, "&gt;");
+}
+
+async function getReferencedIssue() {
+  const response = await octokit.request(`GET /repos/${argv.repo}/issues/${argv.number}`,);
+  return response.data;
+}
+
+// Identify web-features based on spec URLs in the issue body.
+function gatherFeaturesFromSpecUrls(issueBody, gatheredFeatures) {
+  // Find URLs in the issue body.
+  const urls = issueBody.match(/https?:\/\/[^)\s]+/g) || [];
```

Comment:
It feels like a better factoring here would be to have a function that returns URL objects for all the URLs in the issue and then have some functions that try to extract features from the list of URLs, rather than having each function do its own parsing.